### ISSUE TITLE
Unable to add custom note to ELF binary

### ISSUE THREAD

**Issue by mackncheesiest** [lief-project/LIEF#300]
**Describe the bug**
LIEF segfaults when printing an ELF binary dump after adding a note to it
Also, the resulting note is not included in the resulting binary when viewed with `readelf -n`

**To Reproduce**
```
import lief
mybin = lief.parse("./original_elf.bin")
mynote = lief.ELF.Note("CustomNote", lief.ELF.NOTE_TYPES.UNKNOWN, [1, 2, 3])
mynote = mybin.add(mynote)
mybin.write("./modified_elf.bin")
print(mybin)
```

**Expected behavior**
- The new note should be included in the resulting ELF binary
- The new note should be included in the python console dump (and not cause a segfault)

**Environment (please complete the following information):**
 - System and Version: Ubuntu 18.04 LTS
 - Target format: ELF
 - LIEF commit version: fa3195b1bdb34ee126cc9845afca5b3c13834e79

**Additional context**
[Discussion on gitter](https://gitter.im/lief-project/Help?at=5cf6bb414dfb4240047c7fc9)

**Comment by mackncheesiest** [comment]
Currently working on a fix for this in my repository fork

**Comment by mackncheesiest** [comment]
Closed in #301
